tests/test_regression.py

```diff
@@ -0,0 +1,124 @@
+    yaml_file = f"data/yamldb/{theoryid}/{dataset}.yaml"
+    _download_resources(yaml_file, tmp_path)
+    # However, the yaml file goes into the "root" of the yaml directory, move it there!
+    right_yaml = tmp_path / "data" / "yamldb" / f"{dataset}.yaml"
+    (tmp_path / yaml_file).rename(right_yaml)
+    # Download the relevant grids for this dataset
+    res = safe_load(right_yaml.open("r", encoding="utf-8"))
+    grids = [i for j in res["operands"] for i in j]
+    for grid_name in grids:
+        _download_resources(f"data/grids/400/{grid_name}.pineappl.lz4", tmp_path)
+    return grids
+
+
+class _FakePDF:
+    """A Fake lhapdf-like PDF to evolve the grids"""
+
+    def __init__(self):
+        pids = np.arange(-6, 7, dtype=int)
+        pids[6] = 21
+
+        alphas = np.linspace(1.2, 1.8, len(pids))
+        betas = np.linspace(1.2, 3.8, len(pids))
+
+        self._alphas = dict(zip(pids, alphas))
+        self._betas = dict(zip(pids, betas))
+
+    def xfxQ2(self, pid, x, q2):
+        """Compute x^alpha*(1-x)^beta"""
+        alpha = self._alphas[pid]
+        beta = self._betas[pid]
+        return np.power(x, alpha) * np.power(1 - x, beta)
+
+
+def _trim_template(template_card, take_points=10):
+    """Trim the template card so that the number of x-values to compute is much smaller"""
+    card_info = safe_load(template_card.open("r", encoding="utf-8"))
+    original_x = np.array(card_info["rotations"]["xgrid"])
+    skip = int(len(original_x) / take_points)
+    idxs = list(range(0, len(original_x), skip))
+    card_info["rotations"]["xgrid"] = original_x[idxs].tolist()
+    dump(card_info, template_card.open("w", encoding="utf-8"))
+
+
+@pytest.mark.parametrize("dataset", ["LHCBWZMU8TEV", "INTEGXT3"])
+def test_regression(tmp_path, dataset, rebuild=False):
+    """Run pineko through subprocess to ensure that the shell scripts are working exactly
+    as intended.
+    """
+    # We start by downloading pineko.toml in order to generate the folder structure
+    _download_resources("pineko.toml", tmp_path)
+    # Which we create... now!
+    run(["pineko", "scaffold", "new"], cwd=tmp_path, check=True)
+
+    # Now download other necessary objects
+    _download_resources(f"data/theory_cards/{THEORYID}.yaml", tmp_path)
+    template_card = _download_resources(
+        f"data/operator_cards/{THEORYID}/_template.yaml", tmp_path
+    )
+    _trim_template(template_card)
+
+    # And use some small (but not trivial!) dataset to test
+    gridnames = _download_dataset(dataset, THEORYID, tmp_path)
+
+    # Now go, first with eko creation
+    run(
+        ["pineko", "theory", "opcards", str(THEORYID), dataset],
+        cwd=tmp_path,
+        check=True,
+    )
+    run(["pineko", "theory", "ekos", str(THEORYID), dataset], cwd=tmp_path, check=True)
+
+    # Then FK Table production!
+    run(["pineko", "theory", "fks", str(THEORYID), dataset], cwd=tmp_path, check=True)
```

Comment:
See above :)

I'm asking this also because it makes easier refactoring. If ever we will change something, the static analyzer knows functions, and can help us to update them consistently.
These instead are just strings, and it makes harder to discover them ("pineko", "theory", "ekos", and "fks" are used all over the place...).

Reply:
This is exactly what I want. If that happens then I know that all the documentation / my own bash scripts / etc are going to break!

Reply:
Not a fan of doing two tasks in a single shot: you're testing numbers here, so I'd test the CLI somewhere else.

But if you believe to be better like this, because it reproduces a full user run, I'll accept your point of view :)

Reply:
I'm testing numbers because is the only thing that came to mind, but they are not the point. It is an "user run test" indeed.